dbt/schema_tester.py

```diff
@@ -0,0 +1,174 @@
+import os
+import yaml
+
+from dbt.runner import RedshiftTarget
+
+QUERY_VALIDATE_NOT_NULL = """
+with validation as (
+  select "{field}" as f
+  from "{schema}"."{table}"
+)
+select count(*) from validation where f is null
+"""
+
+QUERY_VALIDATE_UNIQUE = """
+with validation as (
+  select "{field}" as f
+  from "{schema}"."{table}"
+)
+select count(*) from (
+  select f from validation group by f having count(*) > 1
+)
+"""
+
+QUERY_VALIDATE_REFERENTIAL_INTEGRITY = """
+with parent as (
+  select "{parent_field}" as id
+  from "{schema}"."{parent_table}"
+), child as (
+  select "{child_field}" as id
+  from "{schema}"."{child_table}"
+)
+select count(*) from child
+where id not in (select id from parent) and id is not null
+"""
+
+class SchemaTester(object):
+    def __init__(self, project):
+        self.project = project
+
+    def project_schemas(self):
+        schemas = {}
+
+        for source_path in self.project['source-paths']:
+            full_source_path = os.path.join(self.project['project-root'], source_path)
+            for root, dirs, files in os.walk(full_source_path):
+                for filename in files:
+                    if filename == "schema.yml":
+                        filepath = os.path.join(root, filename)
+                        abs_path = os.path.join(root, filename)
+                        rel_path = os.path.relpath(abs_path, full_source_path)
+                        parent_path = os.path.dirname(rel_path)
+                        with open(filepath) as fh:
+                            project_cfg = yaml.safe_load(fh)
+                            schemas[parent_path] = project_cfg
+
+        return schemas
+
+    def get_query_params(self, table, field):
+        target_cfg = self.project.run_environment()
+        schema = target_cfg['schema']
+        return {
+            "schema": schema,
+            "table": table,
+            "field": field
+        }
+
+    def make_query(self, query, params):
+        return query.format(**params)
+
+    def get_target(self):
```

Comment:
This seems weird. Why are we doing this in this specific file? Why isn't this being done at some more global level that can be used in testing and execution?